pype_schema/tag.py

```diff
@@ -305,7 +331,7 @@ def __init__(
 
             if determine_type:
                 if tag_type is not None:
-                    if tag_type != tag.tag_type:
+                    if not tag.check_type_compatibility(tag_type):
```

Comment:
I believe there is a bug in this logic. on L315 `determine_type` is only set to `True` if `tag_type is None`. This means that:

```
if determine_type:
    if tag_type is not None: 
        ...
    else:
        ...
```

is a tautology. I.e., you will always enter the `else` portion of the statement